File: PJS5_POC_Cpp/Sources/Weapon/ThermalWeapon/ThermalWeapon.cpp
#include "ThermalWeapon.h"

namespace WEAPON
{
	THERMAL_WEAPON::THERMAL_WEAPON(const int iId, const std::string& sName, const int iArmor, const int iLifePoint, const int iDamage, const int iPowerConsumption, const int iAccuracy, const int iMinAccuracy, const int iFuelBurn)
		: IWEAPON::IWEAPON(iId, sName, iArmor, iLifePoint, iDamage, iPowerConsumption, iAccuracy, iMinAccuracy, WEAPON_TYPE::THERMAL_WEAPON),
		_iFuelBurn(iFuelBurn)
	{
	}
}

File: PJS5_CSharp/PJS5_CSharp/Sources/Weapon/ThermalWeapon/ThermalWeapon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WEAPON;

namespace PJS5_CSharp.Sources.Weapon.ThermalWeapon
{
    public class THERMAL_WEAPON : IWeapon
    {
        private int _iFuelBurn = -1;

        public THERMAL_WEAPON(int iId, string sName, int iArmor, int iLifePoint, int iDamage, int iPowerConsumption, int iAccuracy, int iMinAccuracy, int iFuelBurn)
            : base(iId, sName, iArmor, iLifePoint, iDamage, iPowerConsumption, iAccuracy, iMinAccuracy, WEAPON_TYPE.THERMAL_WEAPON)
        {
            _iFuelBurn = iFuelBurn;
        }

        public override int GetSpecificity()
        {
            return _iFuelBurn;
        }
    }
}
